src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,146 @@
+import { useEffect, useState, useRef } from 'react';
+import { AnimatePresence, Reorder, motion } from 'framer-motion';
+import { Todo } from '../../types/Todo';
+import { Filter } from '../../types/Filter';
+import { IconDelete } from '../UI/IconDelete';
+import { IconEdit } from '../UI/IconEdit';
+import { Checkbox } from '../UI/Checkbox';
+import styles from './TodoItem.module.scss';
+
+type Props = {
+  todo: Todo,
+  todos: Todo[],
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>,
+  setFilter: React.Dispatch<React.SetStateAction<Filter>>,
+  handleToggleTodo: (id: string) => void,
+  handleDeleteTodo: (id: string) => void,
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  todos,
+  handleToggleTodo,
+  handleDeleteTodo,
+  setTodos,
+  setFilter,
+}) => {
+  const { id, text, completed } = todo;
+  const [isCompleted, setIsCompleted] = useState(completed);
+  const [isEditTodo, setIsEditTodo] = useState(false);
+  const [newText, setNewText] = useState(text);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [isEditTodo]);
+
+  useEffect(() => {
+    setTimeout(() => {
+      setIsCompleted(completed);
+    }, 1000);
+  }, [completed]);
+
+  const handleEditTodo = () => {
+    setIsEditTodo(false);
+
+    if (newText === text || !newText.trim()) {
+      setNewText(text);
+
+      return;
+    }
+
+    setFilter(Filter.All);
+
+    setTodos(todos.map(t => {
```

Comment:
Don't cut variable naming for better readability
```suggestion
    setTodos(todos.map(todo => {
```